lib/req_bigquery.ex

```diff
@@ -178,8 +203,40 @@ defmodule ReqBigQuery do
     }
   end
 
+  defp rows_stream(initial_response, request_options) do
+    Stream.unfold({:initial, initial_response}, fn
+      {:initial, %{"rows" => rows} = initial_body} ->
+        {rows, initial_body}
+
+      %{
+        "pageToken" => page_token,
+        "jobReference" => %{"jobId" => job_id, "projectId" => project_id}
+      } ->
+        resp = page_request(request_options, project_id, job_id, page_token)
+        {resp.body["rows"], resp.body}
+
+      _end ->
+        # last iteration didn't have pageToken
+        nil
+    end)
+    |> Stream.flat_map(& &1)
+  end
+
+  defp page_request(options, project_id, job_id, page_token) do
+    uri =
+      URI.parse(
+        "#{@base_url}/projects/#{project_id}/queries/#{job_id}?maxResults=#{options.max_results}&pageToken=#{page_token}"
+      )
+
+    token = Goth.fetch!(options.goth).token
+
+    Req.new(url: uri)
+    |> Request.merge_options(auth: {:bearer, token})
```

Comment:
blast from the past :) any reason this wasn't `Req.new(url: url, auth: {:bearer, token})`?

Reply:
Nothing in particular as i can remember